analysis/mage/src/StandardChecks.tsx

```diff
@@ -0,0 +1,161 @@
+        }
+      });
+    return castArray;
+  }
+
+  countTotalCasts(spell: SpellInfo) {
+    return this.abilityTracker.getAbility(spell.id).casts;
+  }
+
+  countEventsDuringBuff(buff: SpellInfo, eventType: string, cast?: SpellInfo | undefined) {
+    /**
+     * @param buff the spell object for the buff
+     * @param eventType the type of event that you want to search for. i.e. "cast", "begincast", EventType.Cast, EventType.BeginCast, etc.
+     * @param cast an optional cast spell object to count. Omit or leave undefined to count all casts
+     */
+    const events = this.getEventsDuringBuff(buff, eventType, cast);
+    return events.length;
+  }
+
+  getEventsDuringBuff(buff: SpellInfo, eventType: string, spell?: SpellInfo | undefined) {
+    /**
+     * @param buff the spell object for the buff
+     * @param eventType the type of event that you want to search for. i.e. "cast", "begincast", EventType.Cast, EventType.BeginCast, etc.
+     * @param spell an optional spell object to search. Omit or leave undefined to count all events
+     */
+    const buffRemovals: AnyEvent[] = this.eventHistory.last(
+      undefined,
+      undefined,
+      Events.removebuff.by(SELECTED_PLAYER).spell(buff),
+    );
+    if (this.selectedCombatant.hasBuff(buff.id, this.owner.fight.end_time - 1)) {
+      buffRemovals.push(this.eventHistory.last(1, undefined, Events.fightend)[0]);
+    }
+    let events: any;
+    const filteredEvents: any = [];
+    buffRemovals &&
+      buffRemovals.forEach((e) => {
+        const buffApply = this.eventHistory.last(
+          1,
+          undefined,
+          Events.applybuff.by(SELECTED_PLAYER).spell(buff),
+          e.timestamp,
+        )[0];
+        const buffDuration = e.timestamp - buffApply.timestamp - 1;
+        switch (eventType) {
+          case EventType.Cast:
+            events = this.getCastEvents(undefined, e.timestamp, buffDuration, spell);
+            break;
+          case EventType.BeginCast:
+            events = this.getBeginCastEvents(undefined, e.timestamp, buffDuration, spell);
+            break;
+        }
+        events &&
+          events.forEach((event: AnyEvent) => {
+            filteredEvents.push(event);
+          });
+      });
+    return filteredEvents;
+  }
+
+  getCastEvents(
+    count?: number | undefined,
+    startTimestamp?: number | undefined,
+    duration?: number | undefined,
+    spell?: SpellInfo | undefined,
+  ) {
+    const events = this.eventHistory.last(
+      count,
+      duration,
+      spell ? Events.cast.by(SELECTED_PLAYER).spell(spell) : Events.cast.by(SELECTED_PLAYER),
+      startTimestamp || this.owner.fight.end_time,
+    );
+    const filteredCasts: CastEvent[] = [];
+    events &&
+      events.forEach((cast) => {
+        if (CASTS_THAT_ARENT_CASTS.includes(cast.ability.guid)) {
+          return;
+        } else {
+          filteredCasts.push(cast);
+        }
+      });
+    return filteredCasts;
+  }
+
+  getBeginCastEvents(
```

Comment:
It's not clear to me that this needs to be provided as a 'standard check' - it seems like it's just using `eventHistory.last` with the type locked as `BeginCastEvent`. Why not just use `eventHistory.last`?